### DISCUSSION TITLE
ISSUE WITH DSM GENERATOR

### DISCUSSION THREAD

**Discussion by buritilg** [UMEP-dev/UMEP#500] [General]
I am not able to generate the DSM from DEM and building footprints. Could somebody help me?  I am new to UMEP. Please take a look at the attached DEM and building footprint files. The CRS is the same for both layers.

2023-02-07T12:13:59     WARNING    Traceback (most recent call last):
               line 645, in start_progress
              dsm_array = np.array(dsm_raster.ReadAsArray().astype(np.float))
             AttributeError: 'NoneType' object has no attribute 'ReadAsArray'

[DEM_NAD83SOUTH.zip](https://github.com/UMEP-dev/UMEP/files/10677732/DEM_NAD83SOUTH.zip)
[Building_footprints.zip](https://github.com/UMEP-dev/UMEP/files/10677762/Building_footprints.zip)

**Comment by biglimp** [comment]
Hi. We have a discussion forum that is probably more effective if you want help. I will move you over there. Maybe @nilswallenberg can take a look tomorrow…

**Comment by nilswallenberg** [comment]
Hi,

I was able to produce a DSM with your data (in UMEP and UMEP processing). Maybe you can provide a screenshot of the tool window with settings used.

Best,
Nils

**Reply by buritilg** [reply]
@nilswallenberg thank you for checking my files. I still cannot produce the DSM. I appreciate your help.
![image](https://user-images.githubusercontent.com/124711187/217566097-779a715f-7583-48b0-a37d-974ac539c0ad.png)

**Reply by nilswallenberg** [reply]
Did you set the extent of the data? Try Extent from layer = your DEM-layer.

**Reply by biglimp** [reply]
@nilswallenberg , maybe add a warning in the tool that you have to set the extent?

**Reply by buritilg** [reply]
Thank you so much! It worked now. I agree the warning might be helpful!

Best,

**Comment by biglimp** [comment]
I just tried and was also successful.

A comment: Your DEM is not a DEM but a DSM, i.e. including buildings and vegetation. 

![image](https://user-images.githubusercontent.com/11908356/217574321-a474f8c0-36ce-4c75-a236-f2ab6275c840.png)

**Comment by buritilg** [comment]
Yeah, the building elevation numbers are not adding up as they should. I extracted the DEM and DSM from LiDAR data by following an ArcGIS pro workflow (see the link below). I will double-check my process and see what is wrong with it. I am also looking into the LAStool QGIS plugging. 

https://desktop.arcgis.com/en/arcmap/latest/manage-data/las-dataset/lidar-solutions-creating-raster-dems-and-dsms-from-large-lidar-point-collections.htm

**Comment by buritilg** [comment]
[TRIAL_0209.zip](https://github.com/UMEP-dev/UMEP/files/10702526/TRIAL_0209.zip)
Hello @biglimp  and @nilswallenberg, I appreciated your help and patience yesterday! Sorry for bothering again. I am still learning how to use the tool. I was finally able to clean up the DEM and DSM files. But now the sky view factor is taking too long to run. Do you have any tips on how I could optimize the processing time? 

![image](https://user-images.githubusercontent.com/124711187/217967269-a11c992d-e9bc-41eb-b5fb-4dbe6000e102.png)

**Reply by nilswallenberg** [reply]
From what I can see you have done what you can so far (use 153 shadow images instead of 655), if you don't want to decrease the pixel size of your raster data or decrease the extent of your area of interest. What is the pixel size of your raster data and how many rows (length) and columns (width) do you have?

**Comment by buritilg** [comment]
@nilswallenberg Thank you for your reply. I waited for the PC to run for a few hours and got a decent mean radiant temperature temp output. But now, I am upscaling my pilot study to simulate the thermal conditions for an entire neighborhood. I created a mosaic of the DEMs before running the sky view factors, and I realize it is taking exponentially more time to process. Is it better to mosaic after running UMEP SOLWEIG outputs? Does the mosaic order affect any of the temperature/ sky view factor estimates? 

I am checking the 153 shadow images option. 

![image](https://user-images.githubusercontent.com/124711187/220979687-832b54b3-e530-4ff7-b83d-3b78d03a5c75.png)
